docs/blocks/blocks.md

```diff
@@ -35,6 +35,39 @@ Blocks must be [registered][registering] to function.
 
     When a block is registered, *only* a block is registered. The block does not automatically have an `ItemBlock`. To create a basic `ItemBlock` for a block one should use `new ItemBlock(block).setRegistryName(block.getRegistryName())`. The unlocalized name is the same as the block's. Custom subclasses of `ItemBlock` may be used as well. Once an `ItemBlock` has been registered for a block, `Item.getItemFromBlock` can be used to retrieve it. `Item.getItemFromBlock` will return `null` if there is no ItemBlock for the Block, so if you are not certain that there is an ItemBlock for the Block you are using, check for null.
 
+Coloring a Block
+----------------
+
+Blocks' textures can be programmatically colored. Many Vanilla blocks utilize this functionality. For instance, color of grass, vines, waterlilys and etc. changes depending on the biome it's placed in.
+
+### Block Color Handlers
+
+A block color handler is required to color a block and is an instance of `IBlockColor`, which adds the following method.
+
+```java
+int colorMultiplier(
+  IBlockState state, 
+  @Nullable IBlockAccess worldIn, 
+  @Nullable BlockPos pos, 
+  int tintIndex)
+```
+
+This method returns a hex representation of a color in an integer. 
+
+The `IBlockState`, `IBlockAccess` and `BlockPos` passed to the method make it possible to change color multiplier dynamically. `BlockModelRenderer#renderModelBrightness(IBakedModel, IBlockState, float, boolean)` passes null `worldIn` and `pos` calling the method.
+
+Tint indexes are specified for faces of an element in block's model JSON file. A face without a tint index won't be colored, and therefore won't have its color handler called. Block's digging and hitting particles are hardcoded to have zeroth tint index.
+
+### Registering a Block Color Handler
+
+Block color handlers must be registered by calling `BlockColors#registerBlockColorHandler(IBlockColor, Block)` to function. Same can be done for muliple blocks. An instance of `BlockColors` can be obtained by calling `Minecraft#getBlockColors()`. 
```

Comment:
Change the method reference to `BlockColors#registerBlockColorHandler(IBlockColor, Block...)`